### PR TITLE
Disable Via 3's signing of Via HTML URLs

### PR DESCRIPTION
**Pull request by marcospri**

### CODE REVIEW

**Review by seanh — APPROVED** [review]

**Comment by seanh** [comment]
Rebased

**Review by seanh — CHANGES_REQUESTED** [review]
On second look I'm not sure this works: I'm still seeing the `via.sec` param on the Via HTML URLs locally after changing LMS and Via 3 to use this.

I wonder if we should simplify this by changing Via 3 to accept requests if the `Referer` header points to either LMS or Via 3, the same as Via HTML does. Then we could simply remove the URL signing entirely from h-vialib's `url_for()` function

**Comment by seanh** [comment]
Ah, I see what's happening: LMS does add `via.sec` when creating the Via _3_ URL (which is correct: Via 3 URLs do still need to be signed currently). Even though signing is disabled when Via 3 then creates the Via _HTML_ URL, the original `via.sec` (the one that LMS created for Via 3) gets passed through to Via HTML. It doesn't do any harm but it might be best to remove that to prevent confusion:

```diff
diff --git a/src/h_vialib/client.py b/src/h_vialib/client.py
index a6c1e4e..d628958 100644
--- a/src/h_vialib/client.py
+++ b/src/h_vialib/client.py
@@ -79,6 +79,9 @@ class ViaClient:  # pylint: disable=too-few-public-methods
         if blocked_for:
             query["via.blocked_for"] = blocked_for
 
+        if "via.sec" in query:
+            del query["via.sec"]
+
         if doc.is_html:
             # Optimisation to skip routing for documents we know are HTML
             via_url = self._url_for_html(doc.url, query)
```

Other than that, I think this does work correctly: Via 3 URLs are still being signed but Via HTML ones aren't

File: tests/unit/h_vialib/client_test.py [hunk 1]
```diff
@@ -96,15 +95,14 @@ def test_url_for_with_html(self, client):
         )
 
     def test_url_for_with_html_and_blocked_for(self, client):
-        url = "http://example.com/path?a=1&a=2"
+        url = "http://example.com/path?a=1&a=2&via.sec=REMOVED"
```
Comment (seanh) [inline tests/unit/h_vialib/client_test.py]:
```suggestion
        url = "http://example.com/path?a=1&a=2&via.sec=THIS_SHOULD_BE_REMOVED"
```

**Review by seanh — CHANGES_REQUESTED** [review]

File: tests/unit/h_vialib/client_test.py (hunk 1 above)
Comment (marcospri) [reply tests/unit/h_vialib/client_test.py]: Oh, much clearer yep :+1:

**Pull request merged**